src/e3/anod/context.py

```diff
@@ -611,11 +611,9 @@ def add_dep(spec_instance: Anod, dep: Dependency, dep_instance: Anod) -> None:
 
         # Look for dependencies
         spec_dependencies = []
-        if (
-            f"{primitive}_deps" in dir(spec)
-            and getattr(spec, f"{primitive}_deps") is not None
-        ):
-            spec_dependencies += getattr(spec, f"{primitive}_deps")
+        deps = getattr(spec, f"{primitive}_deps", None)
```

Comment:
I'm not sure about this one. This hides AttributeError raised in <primitive>_deps. Same in other occurrences below.

That's why usage of dir() was initially done, to raise clear error when the attributes were invalid.

Reply:
I see, thanks. With all cumulated performance changes, reverting this part makes the time to check plans go from 20s to 26s. I think we can find another way that will not hide these errors.

Reply:
Done: the new version does not hide AttributeError exceptions.